libraries/chem-meta/src/parsing-utils/molfile-handler.ts

```diff
@@ -1,159 +1,17 @@
-import {ChemicalTableParser, ChemicalTableParserBase, AtomAndBondCounts} from './chemical-table-parser';
-import {V2K_CONST} from '../formats/molfile-v2k-const';
-import {V3K_CONST} from '../formats/molfile-v3k-const';
-
-const enum MOLFILE_VERSION {
-  V2000 = 'V2000',
-  V3000 = 'V3000',
-}
-
-export class MolfileHandler extends ChemicalTableParserBase implements ChemicalTableParser {
-  constructor(molfile: string) {
-    super(molfile);
-    this.init(molfile);
-  }
-
-  /** Init/reset the state of the handler for a new molfile */
-  public init(molfile: string) {
-    super.init(molfile);
-
-    const molfileVersion = MolfileHandler.determineMolfileVersion(this.file);
-    const isV2K = (molfileVersion === MOLFILE_VERSION.V2000);
-
-    this.parseAtomAndBondCounts = isV2K ? this.parseAtomAndBondCountsV2K : this.parseAtomAndBondCountsV3K;
-
-    this.getAtomBlockIdx = isV2K ? this.getAtomBlockIdxV2K : this.getAtomBlockIdxV3K;
-
-    this.getCountsLineIdx = isV2K ? this.getCountsLineV2KIdx : this.getCountsLineV3KIdx;
-
-    this.shiftIdxToXColumn = isV2K ? this.shiftIdxToXColumnV2K : this.shiftIdxToXColumnV3K;
-
-    this.shiftIdxToAtomType = isV2K ? this.shiftIdxToAtomTypeV2K :
-      this.shiftIdxToAtomTypeV3K;
-
-    this.getBondBlockIdx = isV2K ? this.getBondBlockIdxV2K : this.getBondBlockIdxV3K;
-
-    this.shiftIdxToBondedAtomsPair = isV2K ? this.shiftIdxToBondedAtomsPairV2K : this.shiftIdxToBondedAtomsPairV3K;
-  }
-
-  public static createInstance(file: string): MolfileHandler {
-    if (!this.instance)
-      this.instance = new MolfileHandler(file);
-    return this.instance as MolfileHandler;
-  }
-
-  protected parseAtomAndBondCounts!: () => AtomAndBondCounts;
-  protected getCountsLineIdx!: () => number;
-  protected getAtomBlockIdx!: () => number;
-  protected shiftIdxToXColumn!: (lineStartIdx: number) => number;
-  protected shiftIdxToAtomType!: (idx: number) => number;
-  protected getBondBlockIdx!: () => number;
-  protected shiftIdxToBondedAtomsPair!: (lineStartIdx: number) => number;
-
-  protected parseAtomType(idx: number): string {
-    const begin = idx;
-    const end = this.file.indexOf(' ', begin);
-    return this.file.substring(begin, end);
-  }
-
-  /** Determine whether the file is V2000/V3000, or throw */
-  private static determineMolfileVersion(molfile: string): MOLFILE_VERSION {
-    if (MolfileHandler.validateV3K(molfile))
-      return MOLFILE_VERSION.V3000;
-    else if (MolfileHandler.validateV2K(molfile))
-      return MOLFILE_VERSION.V2000;
+import {MolfileHandlerBase} from './molfile-handler-base';
+import {MolfileV2KHandler} from './molfile-v2k-handler';
+import {MolfileV3KHandler} from './molfile-v3k-handler';
+
+/** Defines the proper handler based on the molfile type  */
+export class MolfileHandler {
```

Comment:
Why have this class at all? Can't we rename MolHandlerBase to MolHandler, and add this static method there? One less class, a bit more clarity